ipn/ipnserver/server.go

```diff
@@ -341,6 +360,30 @@ func (s *Server) connCanFetchCerts(ci *ipnauth.ConnIdentity) bool {
 	return false
 }
 
+// connIsLocalAdmin reports whether ci has administrative access to the local
+// machine, for whatever that means with respect to the current OS.
+//
+// This returns true only on Windows machines when the client user is a
+// member of the built-in Administrators group (but not necessarily elevated).
+// This is useful because, on Windows, tailscaled itself always runs with
+// elevated rights: we want to avoid privilege escalation for certain mutative operations.
+func (s *Server) connIsLocalAdmin(ci *ipnauth.ConnIdentity) bool {
```

Comment:
I might be incorrect here, but should we just put the inverse logic of this into the windows section of the `localAPIPermission` func and get rid of `CallerIsLocalAdmin`?

Reply:
So if they are not a local admin, they are read-only?

Reply:
@bradfitz curious about your thoughts here, I might be missing some nuance.

Reply:
Extending `localAPIPermission` does seem like the right direction. Its signature (returning two bools) is already kinda borderline gross. Adding a third bool would probably be over the line. We could add a new struct (unexported ideally but not required) as the result type.

Reply:
My memory is really fuzzy on this because I wrote this code so long ago. IIRC I initially did write it as you described, but yanked it out for some reason that now escapes me. I'll take another look to see whether I can remember why I believed this was warranted.

Reply:
@tylersmalley I think this is minor enough that, given my comment above, I'd prefer dealing with as a follow-up issue. Does that work for you?

Reply:
Works for me